src/shared/directives/conditional-show/domain/conditional-show.model.ts

```diff
@@ -2,42 +2,70 @@ import { _ as _score } from 'underscore';
 
 import { CaseField } from '../../../domain/definition/case-field.model';
 import { FieldsUtils } from '../../../services/fields/fields.utils';
+import { ConditionParser } from '../services/condition-parser.service';
 
 export class ShowCondition {
 
-  private static readonly AND_CONDITION_REGEXP = new RegExp('\\sAND\\s(?![^"]*"(?:(?:[^"]*"){2})*[^"]*$)', 'g');
-  private static readonly OR_CONDITION_REGEXP = new RegExp('\\sOR\\s(?![^"]*"(?:(?:[^"]*"){2})*[^"]*$)', 'g');
-  private static readonly CONDITION_NOT_EQUALS = '!=';
-  private static readonly CONDITION_EQUALS = '=';
-  private static readonly CONTAINS = 'CONTAINS';
+  public static readonly CONDITION_NOT_EQUALS = '!=';
+  public static readonly CONDITION_EQUALS = '=';
+  public static readonly CONTAINS = 'CONTAINS';
   private static instanceCache = new Map<string, ShowCondition>();
-
-  private orConditions: string[] = null;
-  private andConditions: string[] = null;
+  private static readonly validJoinComparators = ['AND', 'OR'];
+  private static processedList: string[] = [];
+  private conditions = [];
 
   public static addPathPrefixToCondition(showCondition: string, pathPrefix: string): string {
     if (!pathPrefix || pathPrefix === '') {
       return showCondition;
     }
-    if (showCondition.search(ShowCondition.OR_CONDITION_REGEXP) !== -1) {
-      let orConditions = showCondition.split(ShowCondition.OR_CONDITION_REGEXP);
-      orConditions = this.extractConditions(orConditions, pathPrefix);
-      return orConditions.join(' OR ');
+    if (!showCondition) {
+      return showCondition;
+    }
```

Comment:
this logic is big weird to me, returning something not existing.

Reply:
changed the logic